CMSIS/RTOS2/FreeRTOS/Source/cmsis_os2.c

```diff
@@ -878,6 +920,50 @@ uint32_t osThreadEnumerate (osThreadId_t *thread_array, uint32_t array_items) {
 }
 #endif /* (configUSE_OS2_THREAD_ENUMERATE == 1) */
 
+#if (configUSE_OS2_CPU_AFFINITY == 1)
+/*
+  Set processor affinity mask of a thread.
+*/
+osStatus_t osThreadSetAffinityMask (osThreadId_t thread_id, uint32_t affinity_mask) {
+  TaskHandle_t hTask = (TaskHandle_t)thread_id;
+  osStatus_t stat;
+
+  if (IRQ_Context() != 0U) {
+    stat = osErrorISR;
+  }
+  else if (hTask == NULL) {
+    stat = osErrorParameter;
```

Comment:
[osThreadSetAffinityMask documentation](https://arm-software.github.io/CMSIS_6/latest/RTOS2/group__CMSIS__RTOS__ThreadMgmt.html#ga60e2518df0194587a134b7e1836f2470) says:
- osErrorParameter: thread_id is NULL or invalid or affinity_mask is incorrect.

The question here is: how can affinity_mask be incorrect? IMHO, 0 is valid, all bits set is also valid, any other setting is also valid.

Reply:
It is not explicitly documented when an affinity_mask would be incorrect. It could be when selecting a processor that does not exist.